### ISSUE TITLE
Remove eslint rule overrides

### ISSUE THREAD

**Issue by CMTegner** [reactjs/react-autocomplete#141]
Once #55 lands we'll have rackt-flavored eslint configured for this project. To get it working we had to add a bunch of overrides, but the goal is to remove them and be 1:1 with eslint-config-rackt (unless we decide to deviate).

Here's the list of overrides to remove:
- [ ] array-bracket-spacing
- [ ] comma-dangle
- [x] eol-last
- [x] indent
- [x] no-cond-assign
- [x] no-console
- [x] no-extra-semi
- [x] no-mixed-spaces-and-tabs
- [x] no-multiple-empty-lines
- [x] no-unused-vars
- [x] no-var
- [x] object-curly-spacing
- [x] quotes
- [x] semi
- [x] space-before-blocks
- [x] space-before-function-paren

I'm sure some of these can be fixed with `--fix`.

**Comment by sprjr** [comment]
@CMTegner I was going to work on this or the mobile issue #111, but noticed that `npm start` is failing for me now (both node4/npm2 and node6/npm3). Putting this in `node_modules/rackt-cli/webpack.config.js` seemed to resolve it:

```js
resolveLoader: {
  root: path.join(__dirname, 'node_modules')
}
```

The error without adding that code is:

```
     + 1 hidden modules

ERROR in multi async-data
Module not found: Error: Cannot resolve module 'babel-loader' in /Users/stephenrivasjr/Development/react-autocomplete
 @ multi async-data

ERROR in multi custom-menu
Module not found: Error: Cannot resolve module 'babel-loader' in /Users/stephenrivasjr/Development/react-autocomplete
 @ multi custom-menu

ERROR in multi static-data
Module not found: Error: Cannot resolve module 'babel-loader' in /Users/stephenrivasjr/Development/react-autocomplete
 @ multi static-data
webpack: bundle is now VALID.
```

I have never really used webpack, so apologies if this is straightforward or obvious as to what changed / what the issue is. Is this something we should open an issue for with `rackt-cli` or is it introduced recently?

**Comment by CMTegner** [comment]
@sprjr you probably want to updated your globally installed `rackt-cli`, then perhaps `rm -rf node_modules` and reinstall just for good measure.

```sh
node --version && npm --version && rackt version && npm start
v6.0.0
3.8.6
0.6.1

> react-autocomplete@1.0.1 start /Users/christian/git/react-autocomplete
> rackt server

http://localhost:8080/
webpack result is served from /__build__/
content is served from /Users/christian/git/react-autocomplete/examples
...
```

**Comment by CMTegner** [comment]
Btw, I think many of these violations can be fixed by enabling the rules and running `eslint --fix`.

**Comment by sprjr** [comment]
I figured it likely would be, I'll take a stab at this while I hold off on my fix for #111

**Comment by CMTegner** [comment]
I've removed the overrides for every rule except comma-dangle and array-bracket-spacing. Dangling commas is something I've come to appreciate a lot over the past couple of years, as it helps reduce the amount of noise in diffs. array-bracket-spacing is stylistic choice I don't agree with, based on no reason in particular other than how my muscle memory writes arrays.

Closing for now, but I'm sure there are lots of other rules we can add in the future.
